client/state/posts/selectors.js

```diff
@@ -149,6 +151,38 @@ export function getSitePostsForQueryIgnoringPage( state, siteId, query ) {
 }
 
 /**
+ * Returns an array of posts for the posts query, including all known queried
+ * pages, preserving hierarchy. Returns null if no posts have been received.
+ * Hierarchy is represented by `parent` and `items` properties on each post.
+ *
+ * @param  {Object} state  Global state tree
+ * @param  {Number} siteId Site ID
+ * @param  {Object} query  Post query object
+ * @return {?Array}        Hierarchical posts for the post query
+ */
+export const getSitePostsHierarchyForQueryIgnoringPage = createSelector(
+	( state, siteId, query ) => {
```

Comment:
>That seems reasonable. We can always revisit, if we find that we're repeating ourselves.

Indeed, that's another point I had thought of last night, is that it should be no trouble to change the default behavior in the future without it impacting existing implementations.